vita3k/util/src/net_utils.cpp

```diff
@@ -327,4 +332,137 @@ bool socketSetBlocking(int sockfd, bool blocking) {
     return true;
 }
 
+std::string get_web_response(const std::string &url) {
+    auto curl = curl_easy_init();
+    if (!curl)
+        return {};
+
+    curl_easy_setopt(curl, CURLOPT_URL, url.c_str());
+    curl_easy_setopt(curl, CURLOPT_USERAGENT, "Vita3K Emulator");
+    curl_easy_setopt(curl, CURLOPT_FOLLOWLOCATION, true); // Follow redirects
+    curl_easy_setopt(curl, CURLOPT_NOPROGRESS, 1L);
+
+    std::string response_string;
+    const auto writeFunc = +[](void *ptr, size_t size, size_t nmemb, std::string *data) {
+        data->append((char *)ptr, size * nmemb);
+        return size * nmemb;
+    };
+    curl_easy_setopt(curl, CURLOPT_WRITEFUNCTION, writeFunc);
+    curl_easy_setopt(curl, CURLOPT_WRITEDATA, &response_string);
+    long response_code;
+    curl_easy_perform(curl);
+    curl_easy_getinfo(curl, CURLINFO_RESPONSE_CODE, &response_code);
+
+    curl_easy_cleanup(curl);
+
+    if (response_code / 100 == 2)
+        return response_string;
+
+    return {};
+}
+
+std::string get_web_regex_result(const std::string &url, const std::regex &regex) {
+    std::string result;
+
+    // Get the response of the web
+    const auto response = get_web_response(url);
+
+    // Check if the response is not empty
+    if (!response.empty()) {
+        std::smatch match;
+        // Check if the response matches the regex
+        if (std::regex_search(response, match, regex)) {
+            result = match[1];
+        } else
+            LOG_ERROR("No success found regex: {}", response);
+    }
+
+    return result;
+}
+
+uint64_t get_current_time_ms() {
+    return std::chrono::duration_cast<std::chrono::milliseconds>(std::chrono::high_resolution_clock::now().time_since_epoch()).count();
+};
+
+static size_t write_data(void *ptr, size_t size, size_t nmemb, void *stream) {
+    size_t written = fwrite(ptr, size, nmemb, (FILE *)stream);
+    return written;
+}
+
+typedef int (*CurlDownloadCallback)(void *c, long t, long d);
+
+bool download_file(std::string url, const std::string &output_file_path, ProgressCallback progress_callback) {
+    CURL *curl_download = curl_easy_init();
+    if (!curl_download)
+        return false;
+
+    CurlDownloadCallback curl_callback = +[](void *c, long t, long d) {
```

Comment:
```suggestion
    CurlDownloadCallback curl_callback = +[](void *user_data, long bytes_downloaded, long bytes_total) {
```
Give more meaning to the variables